src/api/app/models/event/relationship.rb

```diff
@@ -1,7 +1,30 @@
 module Event
   class Relationship < Base
     self.abstract_class = true
-    payload_keys :description, :who, :user, :group, :project, :package, :role
+    payload_keys :description, :when, :who, :user, :group, :project, :package, :role, :notifiable_id
     shortenable_key :description
+
+    def parameters_for_notification
+      super.merge({ notifiable_type: notifiable_type, notifiable_id: notifiable_id })
+    end
+
+    def any_roles
+      [User.find_by(login: payload['user']) || Group.find_by(title: payload['group'])]
+    end
+
+    def notifiable_type
+      return 'Package' if payload['package']
+
+      'Project'
+    end
+
+    def notifiable_id
+      notifiable = if payload['package']
+                     Package.get_by_project_and_name(payload['project'], payload['package'])
+                   else
+                     Project.get_by_name(payload['project'])
+                   end
+      notifiable&.id
```

Comment:
Did you think about all the types of [links](https://github.com/openSUSE/open-build-service/wiki/Links) here?

Reply:
Not really, no